packages/harpguru-core/src/utils/reduce-fullmatrix-to-viewablematrix/reduce-fullmatrix-to-viewablematrix.ts

```diff
@@ -1,30 +1,103 @@
-import type { HarpFaceMatrix } from 'harpparts'
+import { isChromaticHarpFace, InteractionIds } from 'harpparts'
+import type { HarpFaceMatrices, Interaction } from 'harpparts'
 
+import { trimFullMatrixByColumnBounds } from '../trim-fullmatrix-by-columnbounds'
+import { mapMatrixToRemoveBySiblingInteraction } from '../map-matrix-to-remove-by-sibling-interaction'
+import { isMatchHarpFaceMatrices } from '../ismatch-harpfacematrices'
 import { isPopulatedArray } from '../is-populated-array'
 import type { ColumnBounds } from '../../types'
-import { sliceMatrix } from '../../packages/slice-matrix'
 import { doSparceIdedObjectMatricesMatch } from '../../packages/do-sparce-ided-object-matrices-match'
 import type { IdedObject } from '../../packages/do-sparce-ided-object-matrices-match'
 
+const errorMessage = `
+The interaction matrix is somehow not chromatic while
+the sibling matrix is. This should never be the case.
+`
+
 export const reduceFullMatrixToViewableMatrix = <T extends IdedObject>(
-  prevViewableMatrix: HarpFaceMatrix<T>,
-  fullMatrix: HarpFaceMatrix<T>,
+  prevViewableMatrices: HarpFaceMatrices<T>,
+  fullMatrix: HarpFaceMatrices<T>,
+  fullInteractionMatrix: HarpFaceMatrices<Interaction>,
   columnBounds: ColumnBounds
-): HarpFaceMatrix<T> => {
-  if (columnBounds === 'FIT') {
-    if (doSparceIdedObjectMatricesMatch(prevViewableMatrix, fullMatrix))
-      return prevViewableMatrix
-    return fullMatrix
-  }
+): HarpFaceMatrices<T> => {
+  const nextViewableMatrix = (() => {
+    if (isChromaticHarpFace(fullMatrix)) {
+      if (!isChromaticHarpFace(fullInteractionMatrix)) throw Error(errorMessage)
+      // TODO: This is rubbish. Can I not find a way to just remove the Blow & Draw
```

Comment:
I have tried hard here already, but I really don't like it in this state. Perhaps it's worth another try to find a way to filter an enum list.

Reply:
Raised as techdebt: https://github.com/js-jslog/harpguru/issues/131#issuecomment-985588026